### PR TITLE
[AJ-1766] Split compute profile and autopause out of AzureComputeModal

### PR DESCRIPTION
**Pull request by nawatts**
https://broadworkbench.atlassian.net/browse/AJ-1766

Prefactoring for [AJ-1766](https://broadworkbench.atlassian.net/browse/AJ-1766). We want to limit the available compute profiles and require auto-pause for cloud environments in "Lite" billing projects.

To make those changes easier to test, this extracts compute profile and auto-pause configuration from AzureComputeModal into separate components.

I wasn't sure if IA has adopted JSX yet. If desired, I can update these new components to TSX?

Also, while I left existing AzureComputeModal tests alone, some may be duplicated now that these components have their own individual tests.

[AJ-1766]: https://broadworkbench.atlassian.net/browse/AJ-1766?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/libs/azure-utils.js [hunk 1]
```diff
@@ -35,6 +35,8 @@ export const azureMachineTypes = {
 export const getMachineTypeLabel = (key) =>
   _.has(key, azureMachineTypes) ? `${key}, ${azureMachineTypes[key].cpu} CPU(s), ${azureMachineTypes[key].ramInGb} GBs` : 'Unknown machine type';
 
+export const machineTypeHasGpu = (machineType) => !!azureMachineTypes[machineType]?.hasGpu;
```
Comment (jdcanas) [inline src/libs/azure-utils.js:38]: I know it was like this already, but consider converting this file to TS and typing machineType while you're in here if there's not too much complaining from the existing code

**Review by jdcanas — APPROVED** [review]
Thanks for splitting out these components from the ComputeModals, splitting up these modals has been in our backlog for ages but never prioritized. Little by little is the only way it will get done!

**Comment by jdcanas** [comment]
> I wasn't sure if IA has adopted JSX yet. If desired, I can update these new components to TSX?
We haven't started that conversion, we have other priorities such as support environment page decoupling, refactoring our compute modals to be more modular and better tested, adding more test coverage, and converting more files from JS to TS before looking at that conversion

I'll let this go through without holding it up since its a pre-refactoring PR, but before the functionality is implemented I'd ike to ask if there is a planned backend piece for enforcing these compute constraints in "lite" billing projects? Or is Microsoft OK with a user being able to interact with swagger directly to go around these constraints?

File: src/libs/azure-utils.js (hunk 1 above)
Comment (nawatts) [reply src/libs/azure-utils.js:38]: To avoid holding this PR up, I'll do that in a follow up.

**Comment by nawatts** [comment]
> I'd ike to ask if there is a planned backend piece for enforcing these compute constraints in "lite" billing projects? Or is Microsoft OK with a user being able to interact with swagger directly to go around these constraints?

Not sure I can say there's a _planned_ piece yet since this is still kind of in the discovery phase, but we are definitely aware of that limitation.

**Review by davidangb — APPROVED** [review]
:+1:

do you plan to do the same for the persistent disk section?

**Comment by nawatts** [comment]
Today's meeting was the first I heard about limiting persistent disks, so I hadn't looked at it before. But fortunately, the persistent disk section has already been split out into separate TypeScript components.

**Pull request merged**